`par:` no longer works on Windows

Correction: this is not a transient issue.  `par` is no longer usable on Windows.  I'll open a new issue for the "par is broken" rather than "par should be more robust" bit, though.

_Originally posted by @JasonGross in https://github.com/coq/coq/issues/16698#issuecomment-1286259848_


<!-- Thank you for reporting a bug to Coq! -->

#### Description of the problem

```
# Usage: coqworker.opt --kind=[proof|query|tactic] $args
# got -main-channel
# File ".\src/Curves/Edwards/XYZT/Basic.v", line 124, characters 6-336:
# Error:
# Anomaly
# "Uncaught exception Failure("The spawned process did not connect back in 10.0s")."
# Please report at [http://coq.inria.fr/bugs/.](http://coq.inria.fr/bugs/)
# 
# make: *** [Makefile.coq:793: src/Curves/Edwards/XYZT/Basic.vo] Error 129
```
https://github.com/mit-plv/fiat-crypto/actions/runs/3281473086/jobs/5403532417#step:17:1812

_Originally posted by @andres-erbsen in https://github.com/mit-plv/fiat-crypto/issues/1394#issuecomment-1284232309_

#### Coq Version

8.17+alpha / dev (d1e1d3291d34c8eefdec13152434d4238cfb4e19)

--- a/lib/spawn.ml
+++ b/lib/spawn.ml
@@ -77,8 +77,8 @@ let accept (sr,sw) =
 
 let spawn_sock env prog args =
   let main_sock, main_sock_name = mk_socket_channel () in
-  let extra = [| prog; "-main-channel"; main_sock_name |] in
-  let args = Array.append extra args in
+  let extra = [| "-main-channel"; main_sock_name |] in
+  let args = Array.append [| prog |] (Array.append args extra) in
   prerr_endline ("EXEC: " ^ String.concat " " (Array.to_list args));
   let pid =
     Unix.create_process_env prog args env Unix.stdin Unix.stdout Unix.stderr in

